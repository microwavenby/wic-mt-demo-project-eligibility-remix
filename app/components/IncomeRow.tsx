import { ReactElement } from "react";
import { useTranslation } from "react-i18next";

export type IncomeRowProps = {
  periods: string[];
  householdSize: string;
  incomeForHouseholdSize: {
    [key: string]: string;
  };
};

const IncomeRow = (props: IncomeRowProps): ReactElement => {
  const { periods, householdSize, incomeForHouseholdSize } = props;
  const { t } = useTranslation("common");

  // If the householdSize is not an empty string, then use it
  // to lookup the income amounts for each time period.
  // Otherwise, return a placeholder row.

  return (
    <tr>
      {periods.map((period: string) => (
        <td key={period} data-label={t(`Income.incomePeriods.${period}`)}>
          {householdSize !== "" ? incomeForHouseholdSize[period] : "$XX,XXX"}
        </td>
      ))}
    </tr>
  );
};

export default IncomeRow;
