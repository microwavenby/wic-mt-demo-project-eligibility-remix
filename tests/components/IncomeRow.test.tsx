import { renderWithRouter } from "tests/helpers/setup";
import IncomeRow from "app/components/IncomeRow";

const testProps = {
  periods: ["weekly", "monthly", "annual"],
  householdSize: "1",
  incomeForHouseholdSize: {
    weekly: "$100",
    monthly: "$200",
    annual: "$300",
  },
};

it("should match snapshot when householdSize is not empty string", () => {
  const { container } = renderWithRouter(
    <table>
      <tbody>
        <IncomeRow {...testProps} />
      </tbody>
    </table>
  );
  expect(container).toMatchSnapshot();
});

it("should match snapshot when householdSize is an empty string", () => {
  const { container } = renderWithRouter(
    <table>
      <tbody>
        <IncomeRow {...testProps} householdSize="" />
      </tbody>
    </table>
  );
  expect(container).toMatchSnapshot();
});
