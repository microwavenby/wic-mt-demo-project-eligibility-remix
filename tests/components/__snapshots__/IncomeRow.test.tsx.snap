// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when householdSize is an empty string 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          data-label="Weekly"
          data-testid="Weekly-cell"
        >
          $XX,XXX
        </td>
        <td
          data-label="Monthly"
          data-testid="Monthly-cell"
        >
          $XX,XXX
        </td>
        <td
          data-label="Annual"
          data-testid="Annual-cell"
        >
          $XX,XXX
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should match snapshot when householdSize is not empty string 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          data-label="Weekly"
          data-testid="Weekly-cell"
        >
          $100
        </td>
        <td
          data-label="Monthly"
          data-testid="Monthly-cell"
        >
          $200
        </td>
        <td
          data-label="Annual"
          data-testid="Annual-cell"
        >
          $300
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
